insights/parsers/lssap.py

```diff
@@ -0,0 +1,118 @@
+
+"""
+lssap - Command
+============
+
+This module provides processing for the output of the ``lssap`` command on SAP systems.
+The spec handled by this command inlude::
+
+    "lssap"                     : CommandSpec("/usr/sap/hostctrl/exe/lssap")
+
+Class ``Lssap`` parses the output of the ``lssap`` command.  Sample
+output of this command looks like::
+
+ - lssap version 1.0 -
+ ==========================================
+   SID   Nr   Instance    SAPLOCALHOST                        Version                 DIR_EXECUTABLE
+   HA2|  16|       D16|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D16/exe
+   HA2|  22|       D22|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D22/exe
+   HA2|  50|       D50|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D50/exe
+   HA2|  51|       D51|         lu0417|749, patch 10, changelist 1698137|          /usr/sap/HA2/D51/exe
+
+Examples:
+    >>> lssap = shared[Lssap]
+    >>> lssap.instances()
+    ['D16', 'D22', 'D50', 'D51']
+    >>> ps_info.version('D51')
+    '749, path 10, changelist 1698137'
+    >>> lssap.is_instance()
+    True
+    >>> lssap.is_hana()
+    False
+    >>> lssap.data[3]
+    {'SID': 'HA2', 'Nr': '51', 'Instance': 'D51', 'SAPLOCALHOST': 'lu0417',
+     'Version': '749, path 10, changelist 1698137', 'DIR_EXECUTABLE': '/usr/sap/HA2/D51/exe'}
+    >>> lssap.data[3]['Instance']
+    'D51'
+"""
+from .. import Parser, parser, parse_table
+from insights.parsers import ParseException
+import re
+
+
+class SAPList(Parser):
+    """Base class implementing shared code."""
+
+    @property
+    def instances(self):
+        """list: Returns a list of instances from the Instance column."""
+        return [row["Instance"] for row in self.data if "Instance" in row]
+
+    @property
+    def sid(self):
+        """list: Returns a list of the SIDs from the SID column."""
+        return list(set(row["SID"] for row in self.data if "SID" in row))
+
+    def instance_types(self):
+        types = []
```

Comment:
Please include docstrings for this method. and the other public methods below so that they will be included in the documentation.